src/Microsoft.Azure.Batch.SoftwareEntitlement/TokenGenerator.cs

```diff
@@ -109,32 +109,34 @@ private List<Claim> CreateClaims(NodeEntitlements entitlements)
         {
             var claims = new List<Claim>();
 
-            if (!string.IsNullOrEmpty(entitlements.VirtualMachineId))
-            {
-                _logger.LogDebug("Virtual machine Id: {VirtualMachineId}", entitlements.VirtualMachineId);
-                claims.Add(new Claim(Claims.VirtualMachineId, entitlements.VirtualMachineId));
-            }
-
-            if (!string.IsNullOrEmpty(entitlements.Identifier))
-            {
-                _logger.LogDebug("Entitlement Id: {EntitlementId}", entitlements.Identifier);
-                claims.Add(new Claim(Claims.EntitlementId, entitlements.Identifier));
-            }
+            AddClaim(claims, Claims.VirtualMachineId, "Virtual machine id", entitlements.VirtualMachineId);
+            AddClaim(claims, Claims.CpuCoreCount, "CPU core count", entitlements.CpuCoreCount?.ToString(CultureInfo.InvariantCulture));
+            AddClaim(claims, Claims.BatchAccountId, "Batch account id", entitlements.BatchAccountId);
+            AddClaim(claims, Claims.PoolId, "Pool id", entitlements.PoolId);
+            AddClaim(claims, Claims.JobId, "Job id", entitlements.JobId);
+            AddClaim(claims, Claims.TaskId, "Task id", entitlements.TaskId);
+            AddClaim(claims, Claims.EntitlementId, "Entitlement id", entitlements.Identifier);
 
             foreach (var ip in entitlements.IpAddresses)
             {
-                _logger.LogDebug("IP Address: {IP}", ip);
-                claims.Add(new Claim(Claims.IpAddress, ip.ToString()));
+                AddClaim(claims, Claims.IpAddress, "IP address", ip.ToString());
             }
```

Comment:
Perhaps create an overload 

    void AddClaims(IList<Claim> claims, string claimId, string claimName, IEnumerable<string> claimValue)

Then you could write

    AddClaims(claims, Claims.IpAddress, "IP address", entitlements.IpAddresses.Select(ip => ip.ToString()));
    AddClaims( claims, Claims.Application, "Application Id", entitlements.Applications);

Reply:
Yep, can do